core/client/fs/src/main/java/alluxio/client/file/cache/LocalCacheManager.java

```diff
@@ -120,25 +119,31 @@ private ReadWriteLock getPageLock(PageId pageId) {
   @Override
   public void put(PageId pageId, byte[] page) throws IOException {
     PageId victim = null;
-
+    long victimSize = 0;
+    long pageSize = 0;
     ReadWriteLock pageLock = getPageLock(pageId);
     try (LockResource r = new LockResource(pageLock.writeLock())) {
       boolean alreadyCached;
       boolean needEvict = false;
       try (LockResource r2 = new LockResource(mMetaLock.writeLock())) {
         alreadyCached = mMetaStore.hasPage(pageId);
-        if (!alreadyCached) {
-          needEvict = mPageStore.size() + 1 > mCacheSize;
+        if (alreadyCached) {
+          pageSize = mMetaStore.getPageSize(pageId);
+        } else {
+          needEvict = mPageStore.bytes() + page.length > mCacheSize;
           if (needEvict) {
             victim = mEvictor.evict();
+            victimSize = mMetaStore.getPageSize(victim);
```

Comment:
I think we need a loop here where we evict data, since there is no guarantee one victim is sufficient?

Reply:
As discussed, we will make best effort and evict at most one page to keep put concurrent